app/context.go

```diff
@@ -0,0 +1,100 @@
+package app
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+type ContextKey string
+
+const (
+	TraceIDKey  ContextKey = "trace_id"
+	UserIDKey   ContextKey = "user_id"
+	TenantIDKey ContextKey = "tenant_id"
+)
+
+// Context carries the context of the current execution.
+type Context struct {
+	// original context
+	context.Context
+}
+
+// UserID returns the user id
+func (sc *Context) UserID() (string, bool) {
```

Comment:
Unsure why you return the bool about the presence, you are not doing it for TraceID

Reply:
The idea is that `TraceID()` always exist, so I set it in the New() if it is absent. To allow debugging easily.